app/controllers/lesson_completions_controller.rb

```diff
@@ -1,51 +1,43 @@
 class LessonCompletionsController < ApplicationController
   
-  before_filter :authenticate_request
+  before_action :authenticate_request
+  before_action :lookup_lesson
   
   def create
-    # Validate that the id does in fact correspond to a real lesson
-    # Note that the authentication before_filter makes sure we've got a current_user
-    @lesson = Lesson.where(:id => params[:lesson_id]).first
-
-    if @lesson.nil?
-      render :nothing => true, :status => 400 # bad request
-    else   
-      @course = @lesson.course
-      @next_lesson = @lesson.next_lesson
-      lc = LessonCompletion.new(:student_id => current_user.id, :lesson_id => @lesson.id)
-      if lc.save
-        render "create", :formats => [:js]
-      else
-        render :nothing => true, :status => 400 # bad request
-      end
+    @course = @lesson.course
```

Comment:
@course is only being used in one of the partials. Can we remove this and replace @course in the partial with @lesson.course